app/services/site_configs/upsert.rb

```diff
@@ -0,0 +1,114 @@
+module SiteConfigs
+  module Upsert
+    include AuthenticationHelper
+
+    EMOJI_ONLY_FIELDS = %w[community_emoji].freeze
+    IMAGE_FIELDS =
+      %w[
+        main_social_image
+        logo_png
+        secondary_logo_url
+        campaign_sidebar_image
+        mascot_image_url
+        mascot_footer_image_url
+        onboarding_logo_image
+        onboarding_background_image
+        onboarding_taskcard_image
+      ].freeze
+
+    VALID_URL = %r{\A(http|https)://([/|.|\w|\s|-])*.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?\z}.freeze
+
+    def self.call(configs)
```

Comment:
you could use 

`rails generate service site_configs/upsert config`

which generates a template service, this way you can encapsulate all the methods below in a class with `private` as they all access state, `@configs`, and aren't called anywhere else outside, thus they probably don't need to be class methods

Reply:
I 👍 this suggestion. The reason I added the service generator is to get slightly more consistency in this area, so personally, I'd appreciate it if we start using it a bit more.

Reply:
👍